command/agent/consul/connect.go

```diff
@@ -105,12 +105,9 @@ func connectSidecarRegistration(serviceId string, css *structs.ConsulSidecarServ
 		return nil, err
 	}
 
-	return &api.AgentServiceRegistration{
-		Tags:    helper.CopySliceString(css.Tags),
-		Port:    cMapping.Value,
-		Address: cMapping.HostIP,
-		Proxy:   proxy,
-		Checks: api.AgentServiceChecks{
+	var checks []*api.AgentServiceCheck
+	if len(css.Checks) == 0 {
```

Comment:
With this change, there is no way to actually disable sidecar checks. The job author must specify a `check` or we insert the default TCP ones.  This seems a bit reasonable.